app/controllers/application_controller.rb

```diff
@@ -1,2 +1,26 @@
 class ApplicationController < ActionController::API
+
+  SECRET_KEY = Rails.application.credentials.secret_key
+
+  def encode_token(payload,exp:Time.now+24.hours)
+    payload[:exp] = exp.to_i
+    JWT.encode(payload,'SECRET_KEY')
+  end
+
+  def authorize_admin_token
+    token_header = request.headers["Authorization"]
+    if token_header
```

Comment:
you have code to validate if the token is present.
What if the token is not available in request header.